components/log-viewer-webui/client/src/ui/QueryStatus.jsx

```diff
@@ -6,11 +6,45 @@ import {
 
 import {AxiosError} from "axios";
 
-import {submitExtractIrJob} from "../api/query.js";
+import {submitExtractStreamJob} from "../api/query.js";
 import {QUERY_LOADING_STATES} from "../typings/query.js";
 import Loading from "./Loading.jsx";
 
 
+let enumQueryType;
+/* eslint-disable sort-keys */
+/**
+ * Note: This enum is duplicated from server, as it is non-trivial to include server enums from the
+ * client.
+ *
+ * Enum of job types, matching the `QueryJobType` class in
+ * `job_orchestration.query_scheduler.constants`.
+ *
+ * @enum {number}
+ */
+const QUERY_JOB_TYPE = Object.freeze({
+    SEARCH_OR_AGGREGATION: (enumQueryType = 0),
+    EXTRACT_IR: ++enumQueryType,
+    EXTRACT_JSON: ++enumQueryType,
+});
+/* eslint-enable sort-keys */
+
+/**
+ * Returns the extract job type based on the provided stream type.
+ *
+ * @param {string} streamType
+ * @return {number|null} - The corresponding extract job type, or null if the stream type is not
+ * recognized.
+ */
+const getExtractJobType = (streamType) => {
+    const jobTypeMap = {
```

Comment:
Instead of this getter function, can we define this "map" object as a global variable? e.g.,
```
const EXTRACT_JOB_TYPE = Object.freeze({
    ir: QUERY_JOB_TYPE.EXTRACT_IR,
    json: QUERY_JOB_TYPE.EXTRACT_JSON,
});
```

Then we can retrieve the mapped value by
```
        const extractJobType = getExtractJobType(streamType);
        if ("undefined" === typeof extractJobType) {
            // handle invalid mapped value.
        }
```